modules/publisher/api.py

```diff
@@ -125,6 +125,10 @@ def get_snap_info(snap_name):
         headers=get_authorization_header()
     )
 
+    if response.status_code == 404:
+        message = 'Snap not found: {snap_name}'.format(**locals())
+        flask.abort(404, message)
```

Comment:
Not sure if it's the API call that should call `flask.abort` or should view do `flask.abort` based on 404 response from API. Because maybe we would want to treat 404 from APIs different in view in future (or in other view).